feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,753 @@
+			}
+		}
+	}
+}
+
+func TestInterfaceStatus(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	interfaces := sortedInterfaces(dut.Ports())
+	t.Logf("Interfaces: %s", interfaces)
+	for _, intf := range interfaces {
+		query := gnmi.OC().Interface(intf).State()
+		root := gnmi.Get(t, dut, query)
+		t.Logf("INFO: Interface %s: \n", intf)
+		if root.OperStatus == oc.Interface_OperStatus_NOT_PRESENT {
+			t.Errorf("ERROR: Oper status is not present")
+		} else {
+			t.Logf("INFO: Oper status: %s", root.GetOperStatus())
+		}
+		if root.AdminStatus == oc.Interface_AdminStatus_UNSET {
+			t.Errorf("ERROR: Admin status is not set")
+		} else {
+			t.Logf("INFO: Admin status: %s", root.GetAdminStatus())
+		}
+		if root.Type == oc.IETFInterfaces_InterfaceType_UNSET {
+			t.Errorf("ERROR: Type is not present")
+		} else {
+			t.Logf("INFO: Type: %s", root.GetType())
+		}
+		if root.Description == nil {
+			t.Errorf("ERROR: Description is not present")
+		} else {
+			t.Logf("INFO: Description: %s", root.GetType())
+		}
+		if root.GetCounters().OutOctets == nil {
+			t.Errorf("ERROR: Counter OutOctets is not present")
+		} else {
+			t.Logf("INFO: Counter OutOctets: %d", root.GetCounters().GetOutOctets())
+		}
+		if root.GetCounters().InMulticastPkts == nil {
+			t.Errorf("ERROR: Counter InMulticastPkts is not present")
+		} else {
+			t.Logf("INFO: Counter InMulticastPkts: %d", root.GetCounters().GetInMulticastPkts())
+		}
+		if root.GetCounters().InDiscards == nil {
+			t.Errorf("ERROR: Counter InDiscards is not present")
+		} else {
+			t.Logf("INFO: Counter InDiscards: %d", root.GetCounters().GetInDiscards())
+		}
+		if root.GetCounters().InErrors == nil {
+			t.Errorf("ERROR: Counter InErrors is not present")
+		} else {
+			t.Logf("INFO: Counter InErrors: %d", root.GetCounters().GetInErrors())
+		}
+		if root.GetCounters().InUnknownProtos == nil {
+			t.Errorf("ERROR: Counter InUnknownProtos is not present")
+		} else {
+			t.Logf("INFO: Counter InUnknownProtos: %d", root.GetCounters().GetInUnknownProtos())
+		}
+		if root.GetCounters().OutDiscards == nil {
+			t.Errorf("ERROR: Counter OutDiscards is not present")
+		} else {
+			t.Logf("INFO: Counter OutDiscards: %d", root.GetCounters().GetOutDiscards())
+		}
+		if root.GetCounters().OutErrors == nil {
+			t.Errorf("ERROR: Counter OutErrors is not present")
+		} else {
+			t.Logf("INFO: Counter OutErrors: %d", root.GetCounters().GetOutErrors())
+		}
+		if root.GetCounters().InFcsErrors == nil {
+			t.Errorf("ERROR: Counter InFcsErrors is not present")
+		} else {
+			t.Logf("INFO: Counter InFcsErrors: %d", root.GetCounters().GetInFcsErrors())
+		}
+	}
+}
+
+func TestInterfacesubIntfs(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	interfaces := sortedInterfaces(dut.Ports())
+	t.Logf("Interfaces: %s", interfaces)
+	for _, intf := range interfaces {
+		subIntfIndexes := gnmi.LookupAll(t, dut, gnmi.OC().Interface(intf).SubinterfaceAny().Index().State())
```

Comment:
t.Run() for both interface and subinterface

Reply:
Thanks for your comments. Add t.Run():

```go
	t.Logf("Interfaces: %s", interfaces)
	for _, intf := range interfaces {
		t.Run(intf, func(t *testing.T) {
```

```go
				t.Run(subIntf, func(t *testing.T) {
					intfState := gnmi.Get(t, dut, gnmi.OC().Interface(intf).State())
```